src/parser/csql_grammar.y

```diff
@@ -3155,6 +3160,24 @@ create_stmt
 		DBG_PRINT}}
 	;
 
+class_name_for_synonym
+	: class_name
+		{ $$ = $1; }
+	| class_name_with_server_name
+		{
+		  PT_NODE *cname = CONTAINER_AT_0 ($1);
+		  PT_NODE *sname = CONTAINER_AT_1 ($1);
+
+		  cname->info.name.original = pt_append_string (this_parser, cname->info.name.original, "@");
+		  cname->info.name.original = pt_append_string (this_parser, cname->info.name.original, sname->info.name.original);
```

Comment:
I think it would be better to do `parser_free_tree(this_parser, sname);`.

Reply:
Good point!
It needs to be fixed.